Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CoordinatedGraphicsScene.cpp

```diff
@@ -67,13 +69,48 @@ void CoordinatedGraphicsScene::paintToCurrentGLContext(const TransformationMatri
         currentRootLayer->setTransform(matrix);
 
     bool sceneHasRunningAnimations = currentRootLayer->applyAnimationsRecursively(MonotonicTime::now());
-    m_textureMapper->beginPainting(flipY ? TextureMapper::FlipY::Yes : TextureMapper::FlipY::No);
-    m_textureMapper->beginClip(TransformationMatrix(), FloatRoundedRect(clipRect));
 
+    WTFBeginSignpost(this, CollectDamage);
+    currentRootLayer->collectDamage(*m_textureMapper);
+    WTFEndSignpost(this, CollectDamage);
+    WebCore::Damage boundsDamage;
+    const auto& frameDamage = ([this, &boundsDamage]() -> const WebCore::Damage& {
+        const auto& damage = lastDamage();
+        if (m_propagateDamage != Damage::ShouldPropagate::No && !damage.isInvalid()) {
+            if (m_settings.unifyDamagedRegions()) {
+                boundsDamage.add(damage.bounds());
+                return boundsDamage;
+            }
+            return damage;
+        }
+        return WebCore::Damage::invalid();
+    })();
+
+    FloatRoundedRect actualClipRect(clipRect);
+    if (m_client && m_settings.useDamagingInformationInCompositing()) {
+        // FIXME: Decide what to do with valid-yet-empty damage (bounds:0,0,0,0).
+        m_client->damageRenderTargets(frameDamage);
+
+        const Damage& damageSinceLastRenderTargetUse = m_client->renderTargetDamage();
```

Comment:
I don't understand this, the names damageRenderTarget and renderTargetDamage are a bit confusing, and they are called one after the other, couldn't it be just one?